x-pack/solutions/security/plugins/security_solution/server/lib/detection_engine/rule_types/eql/eql.ts

```diff
@@ -165,6 +170,14 @@ export const eqlExecutor = async ({
 
       let newSignals: Array<WrappedFieldsLatest<BaseFieldsLatest>> | undefined;
 
+      // @ts-expect-error shard_failures exists in
+      // elasticsearch response v9
+      // needs to be spec needs to be backported
+      // https://github.com/elastic/elasticsearch-specification/pull/3372#issuecomment-2621835599
+      // TODO: remove ts-expect-error when ES lib version is updated
+      const shardFailures = response.shard_failures;
+      logShardFailures(isSequenceQuery, shardFailures, result, ruleExecutionLogger);
```

Comment:
Nit: I'd either name this `maybeLogShardFailures` or move the `if (shardFailures)` outside of the function.

Reply:
Fixed here: https://github.com/elastic/kibana/pull/207396/commits/09caabec1ed410256ee917fd55f6990c831e6353